### ISSUE TITLE
Post save for advanced-pod-relationship field failure.

### ISSUE THREAD

**Issue by anitalee-hoi** [pods-framework/pods#4926]
## Issue Overview

I have create a Pod(e.g. Customer) which contain one of the field(e.g. ‘ID card number’) is Relationship type referring to one of my custom database table(e.g custom_idcard_table). The DB structure of my custom table connecting with my Relationship field are ‘cid'(primary key) and ‘card_number’.

The structure of my custom ID card number DB table is like this:

‘custom_idcard_table’ (i.e. 'cid' is Primary key )

cid | dname | status
-- | -- | --
1 | A846104(4) | 1
2 | B844720(7) | 1
3 | T305721(0) | 1

The structure of my Pod object is like the following:
Customer Pod object
- Name( type: pain text)
- Age( type: plain number)
- ID Card Number ( type:  advanced relationship)
        - Table ID Column: ‘**cid**’
        - Table Index Column: ‘**dname**’
        - Display Field in Selection List: ‘**dname**’

## Expected Behavior

When I go to the edit page for one of the ‘Customer’ post, click ‘Update’ button. The post will be save sucessfully.

while running plugins\pods\classes\Pods.php ( line 1246 )
`$table = $this->api->get_table_info( $object_type, $object, null, null, $last_options );`
The expected value is:
<pre>Array
(
    [object_type] => table
    [type] => table
    [object_name] => custom_idcard_table
    [object_hierarchical] => 
    [table] => custom_idcard_table
    [meta_table] => custom_idcard_table
    [pod_table] => wp_6_pods_custom_idcard_table
    [field_id] => cid ← ** key issue
    [field_index] => dname ← **key issue
    [field_slug] => 
    [field_type] => 
    [field_parent] => 
    [field_parent_select] => 
    [meta_field_id] => id
    [meta_field_index] => dname ← ** key issue
    [meta_field_value] => dname ← ** key issue
    [pod_field_id] => id
    [pod_field_index] => name
    [pod_field_slug] => 
    [pod_field_parent] => 
    [join] => Array
        (
        )
    [where] => 
    [where_default] => 
    [orderby] => `t`.`name`, `t`.`cid` ← ** key issue
    [pod] => 
    [recurse] => 
)</pre>

## Current Behavior

After click ‘Update’ button. The post will return an error message like the following:
`Database Error; SQL: SELECT DISTINCT *, `t`.`id` AS `pod_item_id` FROM `custom_idcard_table` AS `t` WHERE ( `t`.`id` = 2 ); Response: Unknown column 't.id' in 'field list'`

while running plugins\pods\classes\Pods.php ( line 1246 )
$table = $this->api->get_table_info( $object_type, $object, null, null, $last_options );
The current value is:
<pre>
Array
(
    [object_type] => table
    [type] => table
    [object_name] => custom_idcard_table
    [object_hierarchical] => 
    [table] => dpm_cityport
    [meta_table] => custom_idcard_table
    [pod_table] => wp_6_pods_custom_idcard_table
    [field_id] => id ← ** key issue
    [field_index] => name ← ** key issue
    [field_slug] => 
    [field_type] => 
    [field_parent] => 
    [field_parent_select] => 
    [meta_field_id] => id 
    [meta_field_index] => name ← ** key issue
    [meta_field_value] => name ← ** key issue
    [pod_field_id] => id
    [pod_field_index] => name
    [pod_field_slug] => 
    [pod_field_parent] => 
    [join] => Array
        (
        )
    [where] => 
    [where_default] => 
    [orderby] => `t`.`name`, `t`.`id` ← ** key issue
    [pod] => 
    [recurse] => 
)
</pre>

## Possible Solution

Base on my observation and debugging, the reason is the transient value stored inside wp_options for the pod is not correct. I can't find the solution but have an idea for solving this like : Make a function call to re-declare the raw data of the transient. I tried delete all the transient records and let system regenerate the new one. But the value keep showing wrongly.
In addition, I found that ‘Post update’ action is not working. However when ‘Post add‘ action is make. the hold system will work normally again until around 24 hours later. May be find the difference between ‘Post add’ feature and ‘Post update’ feature.

## Steps to Reproduce (for bugs)

1. Dashboard -> ‘Customer’ (Pod class) -> edit one of the posts inside
2. click ‘update’ button
3. sql error mentioned in ‘#Current Behavior’ occured and all ‘Post update’ feature failure
4. make a Post create for ‘Customer’ -> click ‘Publish’ button
5. issue disappear and everything keep normal again
6. around 24 hours later, step 1 repeat again

## Pods Package Export (helpful!)

```
Copy and Paste the JSON Export from **Pods Admin, Migrate: Packages, Export** in WordPress admin here
```

## Todos
- [ ] Tests
- [ ] Documentation
I guess it is caching problem, may be suggest me if the plugin have a function for re-declaring the transient value inside wp_option so as to make Pods::get_table_info() / PodsView::get() return a correct array data. Currently have no idea for doing such thing.

## WordPress Environment
<details>
```
Copy and paste your System Details from **Pods Admin > Settings > Debug Information** in WordPress admin here.

WordPress Version: 4.9.5

PHP Version: 5.6.14

MySQL Version: 5.6.16

Server Software: Apache/2.2.31 (Unix) mod_ssl/2.2.31 OpenSSL/0.9.8j-fips DAV/2 PHP/5.6.14

Your User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3359.139 Safari/537.36

Session Save Path:

Session Save Path Exists: No

Session Save Path Writeable: No

Session Max Lifetime: 1440

Opcode Cache:

Apc: No
Memcached: No
OPcache: No
Redis: No
Object Cache:

APC: No
APCu: No
Memcache: No
Memcached: No
Redis: No
WPDB Prefix: wp_6_

WP Multisite Mode: Yes

WP Memory Limit: 96M

Pods Network-Wide Activated: No

Pods Install Location: /home/jail/gsuser01/gsdev/wp-content/plugins/pods/

Pods Tableless Mode Activated: No

Pods Light Mode Activated: No

Currently Active Theme: Vendor Portal Template

Currently Active Plugins:

All-in-One WP Migration: 6.65
All-in-One WP Migration Multisite Extension: 3.49
All-in-One WP Migration Unlimited Extension: 2.10
Debug Bar: 0.9
Debug Objects: 2.5.0
DK PDF: 1.9.3
Duplicate Post: 3.2.1
Force Login: 5.0
Google Analytics Dashboard for WP (GADWP): 5.1.2.5
Gravity Forms: 2.3.1
Gravity Forms Signature Add-on by ApproveMe: 1.5.2.2
Groups: 2.3.1
Groups Restrict Categories: 1.4.1
H-OI Custom Fields Revisions: 1.3
H-OI Vendor Portal: 1.0
Itthinx Updates: 1.2.0
Mammoth .docx converter: 1.8.0
Members: 2.0.2
OnePress Image Elevator: 2.5.8
Pods - Custom Content Types and Fields: 2.7.1
Print, PDF & Email by PrintFriendly: 3.13.0
Really Simple CAPTCHA: 2.0.1
Smart Slider 3: 3.2.14
TablePress: 1.9
UpdraftPlus - Backup/Restore: 1.14.2
WP Cassify: 2.0.4
WP Crontrol: 1.6.2
WP E-Signature: 1.5.3.4
WP E-Signature Business add-ons: 1.5.3.4
WP Log Viewer: 1.2.1
WP Mail SMTP: 1.2.5
```
</details>

**Comment by quasel** [comment]
sorry for the late reply! 
Is this still an issue? How did you configure the Additional options for the connection to the external table?

**Comment by zamoty** [comment]
I'm having the same problem, any news about the issue?

**Comment by JoryHogeveen** [comment]
@anitalee-hoi @zamoty 
Can you please test PR: #5505 ?

**Comment by Discartyptics** [comment]
FWIW, I was having this exact same issue this morning and manually pushing the fix from PR: #5505 fixed the issue for me.

**Comment by JoryHogeveen** [comment]
Thanks for your feedback @Discartyptics !
